tools/scriptutils.js

```diff
@@ -0,0 +1,150 @@
+
+  await fs.writeFile(outFile, icaljs);
+  console.log("Latest release written to " + outFile);
+}
+
+
+async function get_latest_main(outFile) {
+  let response = await octokit.request("GET /repos/{owner}/{repo}/actions/runs", {
+    workflow_id: "ci.yml",
+    branch: "es6",
+    //branch: "main",
+    status: "success",
+    //exclude_pull_requests: true,
+    //event: "push",
+    owner: "mozilla-comm",
+    repo: "ical.js"
+  });
+
+  let workflows = response.data.workflow_runs;
+
+  workflows.sort((a, b) => {
+    let datea = new Date(a);
+    let dateb = new Date(b);
+
+    return (datea < dateb) - (dateb < datea);
+  });
+
+  let archive_download_url = `https://nightly.link/mozilla-comm/ical.js/actions/runs/${workflows[0].id}/distribution.zip`;
+  console.log(archive_download_url);
+  response = await fetch(archive_download_url);
+  if (!response.ok) {
+    throw new Error(response.status);
+  }
+
+  let buffer = Buffer.from(await response.arrayBuffer());
+  let archive = await yauzl.fromBuffer(buffer);
+
+  let entry;
+  do {
+    entry = await archive.readEntry();
+  } while (entry && entry.fileName == "ical.js");
+
+  if (!entry) {
+    throw new Error("ical.js not found in distribution");
+  }
+
+  let stream = await entry.openReadStream();
+  let writeStream = fsc.createWriteStream(outFile);
+
+  await pipeline(stream, writeStream);
+
+  console.log("Latest main written to " + outFile);
+}
+
+async function performance_downloader() {
+  await Promise.allSettled([
+    get_latest_main("./tools/benchmark/ical_main.cjs"),
+    get_latest_release("./tools/benchmark/ical_release.js")
+  ]);
+}
+
+async function generateZonesFile(tzdbDir) {
+  async function processZone(zoneFile) {
+    let contents = await fs.readFile(zoneFile, "utf-8");
+    let lines = contents.split("\r\n");
+    let vtimezone = lines.slice(lines.indexOf("BEGIN:VTIMEZONE") + 1, lines.indexOf("END:VTIMEZONE")).join("\r\n");
+    return `  register(${JSON.stringify(vtimezone)});`;
+  }
+
+  let tzdbVersion = (await fs.readFile(path.join(tzdbDir, "version"), "utf-8")).trim();
+
+  let lines = [
+    `(function() {`,
+    `  function register(tzdata) { ICAL.TimezoneService.register(ICAL.Component.fromString("BEGIN:VTIMEZONE\\r\\n" + tzdata + "END:VTIMEZONE")) };`,
+    `  ICAL.TimezoneService.IANA_TZDB_VERSION = "${tzdbVersion}";`
+  ];
+
+  let contents = await fs.readFile(path.join(tzdbDir, "zoneinfo", "zones.tab"), "utf-8");
+  for (let line of contents.split("\n")) {
+    let parts = line.split(" ");
+    if (parts.length == 3 && parts[2].length) {
+      lines.push(await processZone(path.join(tzdbDir, "zoneinfo", parts[2] + ".ics")));
+    } else if (parts.length == 1 && parts[0].length) {
+      lines.push(await processZone(path.join(tzdbDir, "zoneinfo", parts[0] + ".ics")));
+    }
+  }
+
+  lines.push("})();");
+
+  return lines.join("\n");
+}
+
+async function get_tzdb_version() {
+  let response = await fetch('https://www.iana.org/time-zones');
```

Comment:
`async` functions sometime delay the execution, without having any benefit.  The delay is caused by the need of the runtime to create many contexts for execution.  If here async does not provide any benefit, I am advice against using it.

Reply:
The problem is usually less to do with using `async` on the function, and more to do with knowing when to use—and more importantly, when **not** to use—the `await` keyword inside of an async function.

If you can accomplish the function’s goal with fewer uses of `await`, that lets the runtime keep moving at top speed.

That said, I don’t think this particular case is worth optimizing.  It’s just a simple timezone fetch.  I don’t see how it could impact performance in any human-detectable way.

Reply:
In this specific case, I don't think `fetch` has a mode where it can be called synchronously. Also agree it doesn't make much of a difference in this file based on its purpose, but good to think about in performance critical code.